### ISSUE TITLE
Panic when launching AWS EC2 instance with wrong volume type

### ISSUE THREAD

**Issue by mbentley** [linuxkit/linuxkit#2389]
**Description**
When I attempt to deploy an AWS instance, I am getting a panic:

```
$ linuxkit -v run aws -disk-size 30 -zone b onap-aws
INFO[0003] Created instance i-0f104ef4981a6b5ea
INFO[0034] Instance i-0f104ef4981a6b5ea is running
DEBU[0034] CreateVolume:
{
  AvailabilityZone: "us-east-1b",
  Size: 30,
  VolumeType: "gp2"
}

INFO[0034] Waiting for volume vol-0ddf8754e2f03d512 to be available
INFO[0049] Attaching volume vol-0ddf8754e2f03d512 to instance i-0f104ef4981a6b5ea
WARN[0050] AWS doesn't stream serial console output.
 Please use the AWS Management Console to obtain this output
 Console ouput will be displayed when the instance has been stopped.
WARN[0050] Waiting for instance to stop...
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x8 pc=0x555ba3552ed8]

goroutine 1 [running]:
main.runAWS(0xc4200100d0, 0x5, 0x5)
        /go/src/github.com/linuxkit/linuxkit/run_aws.go:170 +0xfb8
main.run(0xc4200100c0, 0x6, 0x6)
        /go/src/github.com/linuxkit/linuxkit/run.go:46 +0x170
main.main()
        /go/src/github.com/linuxkit/linuxkit/main.go:88 +0x2c0
```

There are no logs in the EC2 instance console and the instance shows as Stopped.

**Steps to reproduce the issue:**
I can repeat this every time

**Describe the results you received:**
Panic

**Describe the results you expected:**
Instance should start

**Additional information you deem important (e.g. issue happens only occasionally):**
```
# moby version
moby version 0.0
commit: c2b081ed8a9f690820cc0c0568238e641848f58f
# linuxkit version
linuxkit version 0.0
commit: bbf87410024706b4a5d20f92dad34d55ab0f4229
```

Here is my linuxkit yaml file:
https://gogs.att.dckr.org/onap/linuxkit-poc-tools/src/linuxkit-new/onap-aws.yml
```
kernel:
  image: linuxkit/kernel:4.9.41
  cmdline: "console=tty0 console=ttyS0 console=ttyAMA0"
init:
  - linuxkit/init:c7d651da1a5e308c757bc61ce6a41804ea843426
  - linuxkit/runc:2d645aca4eee8d0d1f121177f1b696244219d250
  - linuxkit/containerd:f69bc5b8ec0392e2d34d3fd3bfdb7266ef6d4ed4
  - linuxkit/ca-certificates:e44b0a66df5a102c0e220f0066b0d904710dcb10
onboot:
  - name: sysctl
    image: linuxkit/sysctl:154913b72c6f1f33eb408609fca9963628e8c051
  - name: sysfs
    image: linuxkit/sysfs:3ae01a25583ee37a5ff8b09a0e569cb4bd8cf2e9
  - name: format
    image: linuxkit/format:158d992b7bf7ab984100c697d7e72161ea7d7382
  - name: mount
    image: linuxkit/mount:4fe245efb01384e42622c36302e13e386bbaeb08
    command: ["/usr/bin/mountie", "/var/lib/docker"]
  - name: metadata
    image: linuxkit/metadata:cec86f3e1c260c9eafefa80c262fceb40c182ddf
services:
  - name: sshd
    image: linuxkit/sshd:505a985d7bd7a90f15eca9cb4dc6ec92789d51a0
    binds:
     - /root/.ssh/authorized_keys:/root/.ssh/authorized_keys
#  - name: getty
#    image: linuxkit/getty:797cb79e0a229fcd16ebf44a0da74bcec03968ec
#    env:
#     - INSECURE=true
  - name: rngd
    image: linuxkit/rngd:558e86a36242bb74353bc9287b715ddb8567357e
  - name: dhcpcd
    image: linuxkit/dhcpcd:f3f5413abb78fae9020e35bd4788fa93df4530b7
  - name: ntpd
    image: linuxkit/openntpd:0d7befc79842849d0b88d6c3b64200e340d7cf67
  - name: docker
    image: docker:17.06.0-ce-dind
    capabilities:
     - all
    net: host
    mounts:
     - type: cgroup
       options: ["rw","nosuid","noexec","nodev","relatime"]
    binds:
     - /etc/resolv.conf:/etc/resolv.conf
     - /var/lib/docker:/var/lib/docker
     - /lib/modules:/lib/modules
     - /etc/docker/daemon.json:/etc/docker/daemon.json
     - /var/run:/var/run
    command: ["/usr/local/bin/docker-init", "/usr/local/bin/dockerd"]
  - name: onaplaunch
    image: dtr.att.dckr.org/services/onaplaunch:latest
    binds:
     - /var/run:/var/run
files:
  - path: var/lib/docker
    directory: true
  - path: etc/docker/daemon.json
    contents: '{"debug": true}'
  - path: root/.ssh/authorized_keys
    source: ~/.ssh/mbentley_rsa4096.pub
    mode: "0600"
    optional: true
trust:
  org:
    - linuxkit
    - library
```

**Comment by mbentley** [comment]
This is similar to the panic in https://github.com/linuxkit/linuxkit/issues/2387 but it seems like there may be something else that it is complaining about in this environment.

**Comment by mbentley** [comment]
It looks like this may be related to the creation of my AMI.  It is creating the AMI with a disk type of `gp2` so I had to specify `-disk-type standard` when launching my instance so there isn't validation that the AMI root volume type matched the instance.

**Comment by dave-tucker** [comment]
I've just made a PR to resolve the panic. It was due to no console output being present.
There doesn't seem to be an issue with having multiple EBS volumes of different types attached to the same instance. For example, this works in my test environment on `eu-west1`:

```
 ./bin/linuxkit run aws -disk-size 30 -disk-type standard -zone a aws
```

As for why there is no output, I think it's due to your console line in your YAML file.
Try changing it to `cmdline: "console=ttyS0"`
